planet/scripts/cli.py

```diff
@@ -103,3 +107,37 @@ def init(email, password):
     response = call_and_wrap(clientv1().login, email, password)
     write_planet_json({'key': response['api_key']})
     click.echo('initialized')
+
+@cli.command('quota')
+def quota():
+    '''Print allocation and remaining quota in Sqkm.'''
+    try:
+        with open (os.path.join(os.path.expanduser('~'), '.planet.json')) as f:
+            data=json.load(f)
+        main=requests.get('https://api.planet.com/auth/v1/experimental/public/my/subscriptions', auth=HTTPBasicAuth(data["key"], ''))
+        if main.status_code==200:
+            content=main.json()
+            n=(str(main.json()).count("quota_sqkm"))
+            i=0
+            while i<n:
+                item_id=(content[i])
+                print(" ")
+                print("Allocation Name: "+str(item_id['organization']['name']))
+                print("Allocation active from: "+str(item_id['active_from']).split("T")[0])
+                print("Quota Enabled: "+str(item_id['quota_enabled']))
+                print("Total Quota in SqKm: "+str(item_id['quota_sqkm']))
+                print("Total Quota used: "+str(item_id['quota_used']))
+                if (item_id['quota_sqkm'])is not None:
+                    print("Remaining Quota in SqKm: "+str(float(item_id['quota_sqkm'])-float(item_id['quota_used'])))
+                else:
+                    print("No Quota Allocated")
+                print("")
+                i=i+1
+        else:
+            print("Failed with exception code: "+str(main.status_code))      
+    except Exception as e:
+        #print(e)
+        print("Initialize Planet Client First and try again")
+        subprocess.call(["planet", "init"],shell=False)
```

Comment:
So at first glance, this is a nice bit of user-friendly assistance. However, what about users that use the documented option of `-k` to pass in their API key? In that case, they may legitimately not have a `.planet.json` file, but since we only need their API key to make the quota call that shouldn't matter.

What we should do instead:

1) Allow the user to pass in their API key for the `quota` command, just as they can do with e.g., `planet -k <some key> data`. This means adding a `@click.option()` decorator to the `quota()` function for `-k` option.

2) If the check for `planet.json` throws an `IOError`, before forcing the user to initialize the client we should first check for a passed-in API key. IFF that is not provided, then you could continue with the forced initialization. Consider however, that it may be more user-friendly to simply fail with a message asking them to _either_ provide an API key _or_ initialize the client (since they may prefer one over the other).

Reply:
Resolved